instance/models/server.py

```diff
@@ -486,9 +485,15 @@ def _delete_ssh_key(self) -> None:
         We can safely ignore the command's return code, because we just need to be sure that the key has been removed
         for non-existing server - we don't care about non-existing keys.
         """
-        self.logger.info('Deleting SSH key of "%s" host.', self.public_ip)
-        command = f'ssh-keygen -R {self.public_ip}'
+        # self.public_ip might fail because it calls OpenStack APIs,
+        # so we ignore those failures if they happen
         try:
-            subprocess.Popen(command, shell=True, stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL)
-        except subprocess.SubprocessError as e:
-            self.logger.error('Failed to delete SSH key of "%s" host: %s', self.public_ip, e)
+            ip = self.public_ip
+        except novaclient.exceptions.ClientException:
+            self.logger.warning(
```

Comment:
`self.logger.exception` will be cleaner and do the same job, no?

Reply:
The rationale is the same as above, since `logger.exception` implicitly logs at `ERROR` level.

Reply:
Fixed.